src/main/java/net/smartcosmos/dao/relationships/RelationshipDao.java

```diff
@@ -36,9 +54,19 @@
      */
     void delete(String accountUrn, String urn) throws IllegalArgumentException;
```

Comment:
`delete()` should return `List<RelationshipResponse>`.

Why a list? Under the hood, we are going to use Spring Data in the persistence layer where we just need to define an interface method for the delete operation. Spring Data does the actual implementation for us and will return a list of deleted records. If you make Spring Data return a single result, that would just be the count of deleted records.

And since other REST implementations might want to return more information on deleted records, we will return these information. Although we expect only a single entity, there might be some cases where multiple records match and get deleted, and we don't want the persistence layer to decide which one to return. If there are multiple entities, return all of them.

However, that should usually be a list containing a single item, which doesn't hurt at all.